aopy/postproc/neuropixel.py

```diff
@@ -0,0 +1,246 @@
+    
+    return good_unit_labels, ref_violations*100
+
+def get_high_amplitude_units(preproc_dir, subject, te_id, date, port, amp_thresh=50, start_time=0, end_time=None):
+    '''
+    Identifies which units pass the waveform amplitude threshold.
+    Calculates peak to peak amplitude for each unit across all channels the unit is recorded on, 
+    then ony returns units whose amplitude passes the threshold.
+
+    Args:
+        preproc_dir (str): The directory containing the preprocessed data.
+        subject (str): The subject name.
+        te_id (str): The experiment task entry number.
+        date (str): The date of the experiment.
+        port (int): The port number identifying which probe to look at data from.
+        amp_thresh (float, optional): The amplitude threshold (in microvolts) used to filter out units with low peak-to-peak amplitudes. Defaults to 50.
+        start_time (float, optional): Start time (in seconds) for the time window to consider spikes (default is 0).
+        end_time (float, optional): End time (in seconds) for the time window to consider spikes (default is None, meaning all spikes after `start_time` are considered).
+
+    Returns:
+        tuple: A tuple containing:
+            - good_unit_labels (ngoodunit long list of str): List of unit labels for units with a peak-to-peak amplitude greater than the specified threshold.
+            - amplitudes (ngoodunit): Computed amplitude of each unit.
+            - mean_wfs (ntime, ngoodunit): The mean waveform taken from the channel with the highest peak-to-peak amplitude for each unit that passes the amplitude threshold.
+    '''
+    filename_mc = data.get_preprocessed_filename(subject, te_id, date, 'spike')
+    spike_times = data.load_hdf_group(os.path.join(preproc_dir, subject), filename_mc, f'drive{port}/spikes')
+    ap_metadata = data.load_hdf_group(os.path.join(preproc_dir, subject), filename_mc, f'drive{port}/metadata')
+    waveforms = data.load_hdf_group(os.path.join(preproc_dir, subject), filename_mc, f'drive{port}/waveforms')
+    unit_labels = list(waveforms.keys())
+                
+    # Initialize variables
+    nunits = len(unit_labels)
+    nwf_time = waveforms[str(unit_labels[0])].shape[1] # Number of time points in each waveform.
+    good_unit_labels = []
+    amplitudes = []
+    mean_wfs = []
+    for iunit, unit_lbl in enumerate(unit_labels):
+        if end_time is None:
+            relevant_spike_mask = spike_times[unit_lbl] >= start_time
+        else:
+            relevant_spike_mask = np.logical_and(spike_times[unit_lbl] >= start_time, spike_times[unit_lbl] <= end_time)
+        
+        
+        relevant_wfs = waveforms[str(unit_lbl)][relevant_spike_mask,:,:]
+        cent_wfs = relevant_wfs - np.mean(relevant_wfs, axis=1)[:,None,:] # Center each spike on each channel
+        mean_wf = np.mean(cent_wfs, axis=0)*ap_metadata['bit_volts'][0] # Mean across all spikes for each channel. becomes (ntime, nch) array
```

Comment:
Sorry. I'm now updating metadata so that variable names will be the same as the current ECoG metadata. Can you use `ap_metadata['voltsperbit'][0]`, not `ap_metadata['bit_volts'][0]`

Reply:
I'll update to account for both. Should it actually be renamed `ap_metadata['microvoltsperbit']` ?

Reply:
The current metadata in ECoG lfp uses voltsperbit. So I think we should use it.

Reply:
okay, are you converting to volts? (The current scalar converts to microvolts)

Reply:
Ah, I didn't think about it. Changing scale from microvolts to volts now may be confusing?

Reply:
I think it probably makes sense to just convert everything to voltsperbit and update code now. @leoscholl what do you think?

Reply:
add both!